### PR TITLE
make enable_access_request a around hook to clean it up

### PR DESCRIPTION
**Pull request by grosser**
more reliable / no accidental state / no strange instance variables

@msufa 

/fyi @jonmoter @fneves @sbrnunes

### CODE REVIEW

File: test/mailers/access_request_mailer_test.rb [hunk 1]
```diff
@@ -16,11 +16,7 @@
     let(:role) { Role::DEPLOYER }
     subject { ActionMailer::Base.deliveries.last }
 
-    before do
-      enable_access_request(address_list, prefix)
-    end
-
-    after { restore_access_request_settings }
+    around { |t| enable_access_request address_list, email_prefix: prefix, &t }
```
Comment (msufa) [inline test/mailers/access_request_mailer_test.rb]: It seems that now you cannot override `address_list` inside tests. That's the reason the build for this PR is failing `handles no configured email address`

**Comment by grosser** [comment]
fixed, looks good ?

**Comment by msufa** [comment]
👍

**Comment by andreionut** [comment]
👋 @msufa. How are you keeping?

**Comment by msufa** [comment]
Waiting for proper snowfall @andreionut :) Drop by with the team for an offsite, we'll do some open water skating! All the best from STO!

**Pull request merged**